validator/src/main.rs

```diff
@@ -212,7 +212,9 @@ fn get_rpc_node(
         );
 
         if rpc_peers_blacklisted == rpc_peers_total {
-            retry_reason = if blacklist_timeout.elapsed().as_secs() > 60 {
+            retry_reason = if !blacklisted_rpc_nodes.is_empty()
+                && blacklist_timeout.elapsed().as_secs() > 60
+            {
                 // If all nodes are blacklisted and no additional nodes are discovered after 60 seconds,
                 // remove the blacklist and try them all again
                 blacklisted_rpc_nodes.clear();
```

Comment:
Well, there is no point to `.clear()` `blacklisted_rpc_nodes` if it's empty. On the other hand, it's equally harmless to `.clear()` it if it's empty as original code was doing. But, this `if` clause sets `retry_reason` to `"Blacklist timeout expired"`. That's confusing. So, just don't enter the clause if `blacklisted_rpc_nodes` is `.is_empty()`.

This function is already quite complex and not tested. But according to the above reasoning, this change should be pretty low-risk and free of bugs.